tests/spec/core/highlight-spec.js

```diff
@@ -76,4 +76,93 @@ describe("Core — Highlight", () => {
     const pre = doc.getElementById("test");
     expect(pre.querySelectorAll("span[class^=hljs-]").length).toBe(0);
   });
+
+  it("checks if <pre> content is warpped in <code>", async () => {
+    const ops = {
+      config: makeBasicConfig(),
+      body: `${makeDefaultBody()}<section>
+          <pre>
+            function foo() {
+              alert('foo');
+            }
+          </pre>
+        </section>`,
+    };
+    const doc = await makeRSDoc(ops);
+    const pre = doc.querySelectorAll("pre");
+    expect(pre.querySelectorAll("code").length).toBe(1);
+    expect(pre.querySelectorAll("code[class^='javascript']").length).toBe(1);
```

Comment:
this is behavior that's part of highlightjs, and not your code, so I think this whole test can be removed.

Reply:
But, I am checking if the language detected by highlightjs also gets added into the code element apart from the conventional way of defining it as `<pre class="js">...</pre>` so shouldn't I include this case too ?

Reply:
Ah, ok. Good point. Make sure you use: `[class~='javascript']`.